lang/src/splgen/splgen/contract_generator.py

```diff
@@ -149,7 +149,7 @@ def _resolve(self, fun: Function, globals: Set[Dependency]):
         pack_name, *path, func_name = fun.name.split('.')
 
         if not pack_name in package_cache:
-            with open(f'{pack_name}.json') as pack:
+            with open(f'{SPM_PACKAGES_PATH}/{pack_name}.json') as pack:
```

Comment:
I don't know but we may have a problem here. You are not reading versions from package.json; all packages have versions when we install them with spm-cli.